src/corelibs/U2Core/src/gobjects/MultipleAlignmentObject.cpp

```diff
@@ -737,6 +737,26 @@ void MultipleAlignmentObject::releaseState() {
     }
 }
 
+bool MultipleAlignmentObject::isGapRemovePossible() const {
+    const QList<QVector<U2MsaGap>> &listGapModel = getGapModel();
+    for (const QVector<U2MsaGap>& vector : qAsConst(listGapModel)) {
+        if (vector.size() > 1) {
+            //if we have more than one set of gaps in a row - they are 100% removable
+            return true;
+        }
+    }    
+    for (int i = 0; i < getRowCount(); i++) {
+        //if remaining gaps isn't trailing we can remove them too
+        int gapLength = listGapModel[i].size() == 0 ? 0 : listGapModel[i][0].length;
+        const MultipleAlignmentRow &row = getRow(i);
+        qint64 rowLengthWithoutTrailing = row->getRowLengthWithoutTrailing();
```

Comment:
Gap model never has trailing gaps or U2MsaGaps with zero length. Why the gapModel.isEmpty() check is not enough?

Reply:
In this case we will skip situation when there is no trailing gaps in a row but it have one set of gaps in the middle of it.

Reply:
This method looks very ambiguous: which gaps do you mean

1. Leading: the first gap in the gap model starts with 0
2. Trailing: sum of all gap lengths + core-length < alignment-length
3. Middle gaps: gap model size > 1 || gaps[0].startPos > 0
4. Any visible gap in the view: coreLength < alignmentLength

What kind if gaps do you want to check here? 
I think all removable: leading + middle.
 In this case please rename the method to hasNonTrailingGap and implement it like !gapModel.isEmpty()

?

Reply:
You not right about point 3, because it is possible have row with only middle gaps.
Anyway i updated this method and test, with leading gaps test case.

Reply:
I think 3 is right: you should have either at least 2 gap of the first gap starting with > 0 offset.